File: 2261-all-divisions-with-the-highest-score-of-a-binary-array/all-divisions-with-the-highest-score-of-a-binary-array.cpp
class Solution {
public:
    vector<int> maxScoreIndices(vector<int>& nums) {
        int totalOnes=0, leftZeros=0, leftOnes=0,maxScore=-1;
        vector<int> indices;
        for(int num:nums){
            if(num==1) ++totalOnes;
        }

        for(int i=0;i<=nums.size();++i){
            /*
            //The comparision cases are placed first so is mimics the division starting from one that is empty at the left and one that is empty at the right
            */
            int rightOnes=totalOnes-leftOnes;
            int score=leftZeros+rightOnes;

            if(score>maxScore){
                indices.clear();
                maxScore=score;
                indices.push_back(i);
            }
            else if (maxScore==score){
                indices.push_back(i);
            }

            if(i<nums.size()){
                if(nums[i]==0) ++leftZeros;
                else ++leftOnes; // Since it is binary array there is no other case
            }
            
        }
        return indices;
    }
};

File: 2261-all-divisions-with-the-highest-score-of-a-binary-array/all-divisions-with-the-highest-score-of-a-binary-array.py
class Solution:
    def maxScoreIndices(self, nums: List[int]) -> List[int]:
        # 0's in left and 1's in right
        n = len(nums)
        total_ones = nums.count(1)
        left = 0
        right = total_ones
        score = left + right

        sol = []
        for idx,num in enumerate(nums):
            if score==left+right:
                sol.append(idx)
            elif score<left+right:
                score=left+right
                sol = [idx]
            
            if num==0: left+=1
            else: right-=1

        if n-total_ones == score: sol.append(n)
        elif n-total_ones>score: sol=[n]
        
        return sol